qiskit/providers/ibmq/ibmqjob.py

```diff
@@ -281,7 +303,22 @@ def status(self):
         return self._status
 
     def error_message(self):
-        """Return the error message returned from the API server response."""
+        """Provide details about the reason of failure.
+
+        Calling this method if the job status is ERROR, consumes the job.
+
+        Returns:
+            str: An error report if the job errored or None elsewhere.
+        """
+        self._wait_for_completion()
+        if self.status() is not JobStatus.ERROR:
+            return None
+
+        job_response = self._get_job()
+        results = job_response['qObjectResult']['results']
+        if not self._api_error_msg:
+            self._api_error_msg = self._build_error_report(results)
```

Comment:
Probably minor - but in theory there might be a case where the job errors out at a "general" level, without specific errors in the experiments (as per the `JobStatus` definition), and if that happens it seems the error report will likely fail or not provide meaningful information. Non-blocking as I think in practice it is an unlikely situation, but can you follow up on this case or keep an eye if this happens?

Reply:
I had it on my radar (and I'm still trying to force the situation) but I'll document it in a new issue. :+1: